storage/reads/table.gen.go.tmpl

```diff
@@ -73,36 +72,37 @@ func (t *{{.name}}Table) Do(f func(flux.ColReader) error) error {
 		t.mu.Unlock()
 	}()
 
-	if !t.Empty() {
-		t.err = f(t)
-		for !t.isCancelled() && t.err == nil && t.advance() {
-			t.err = f(t)
+	for !t.isCancelled() && t.err == nil  {
+		cr := t.advance()
+		if cr == nil {
+			break
 		}
+		t.err = f(cr)
+		cr.Release()
```

Comment:
Is it possible for Release to be called before another transformation Retains it?

Reply:
I think the contract for `Do` says that as soon as the function ends it goes out of scope so it shouldn't be possible to have anything like the `consecutiveTransport` as that would break that contract (unless the transport retained it itself).

So I don't think this should cause a race condition unless the passed in function is breaking the contract.